src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1882,10 +1882,99 @@ module T = struct
               match
                 O1trace.sync_thread "validate_transaction_against_staged_ledger"
                   (fun () ->
+                    let get_verification_keys account_ids =
+                      List.fold_until account_ids ~init:Account_id.Map.empty
+                        ~f:(fun acc id ->
+                          let get_vk () =
```

Comment:
No need to nest this function, separating it out will improve indentation a lot.